src/services.cpp

```diff
@@ -107,6 +119,53 @@ bool MapperClass::OctomapProcessPCL(std_srvs::SetBool::Request &req,
     return true;
 }
 
+bool MapperClass::AStarService(pensa_msgs::Astar::Request &req,
+                               pensa_msgs::Astar::Response &res) {
+    octomap::point3d p0(req.origin.x,           req.origin.y,      req.origin.z);
+    octomap::point3d pf(req.destination.x, req.destination.y, req.destination.z);
+    double planning_time;
+    std::vector<Eigen::Vector3d> path, pruned_path;
+    double plan_height;
+    mutexes_.update_map.lock();
+        const bool is_mapping_3d = globals_.octomap.IsMapping3D();
+        if (!is_mapping_3d && !helper::AreDoubleSame(p0.z(), pf.z())) {
+            ROS_ERROR("[mapper]: Mapping in 2D and the waypoints are not at the same height. Cannot compute A*");
+            res.success = false;
+        } else {
+            if (!is_mapping_3d) {
+                plan_height = p0.z();
+                p0.z() = 0.0;
+                pf.z() = 0.0;
+            }
+            res.success = globals_.octomap.Astar(p0, pf, req.prune_result, &planning_time, &path, &pruned_path);
+        }
+    mutexes_.update_map.unlock();
+    if (res.success) {
```

Comment:
this needs a corresponding else, presumably that returns `false`

Reply:
I don't think so... I prefer to `return true` in all services, indicating that the service call was successful. The result of `A*` is already captured in `res.success`.
In any case, I have added a portion to return `planning_time` even when `res.success` is `false`.